doc/source/reference/c-api/array.rst

```diff
@@ -1323,6 +1323,28 @@ User-defined data types
 Special functions for NPY_OBJECT
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. warning::
+
+    When working with arrays or buffers filled with objects NumPy tries to
+    ensure such buffers are filled with ``None`` before any data may be read.
+    However, code paths may existed where an array is only initialized to
+    ``NULL``.
+    NumPy itself accepts ``NULL`` as an alias for ``None``, but may ``assert``
+    non-``NULL`` when compiled in debug mode.
+
+    Users **must** expect a value of ``NULL`` when working with buffers created
```

Comment:
An explanation of *why* users need to do this would be helpful.

Reply:
Hmm maybe lead with "Because of this ..."?   The first paragraph is meant to be the reason.

Reply:
Something along the line "Because NumPy is not yet consistent about initialization with ``NULL``, users must ..."

Reply:
Ah, nice!  Adopted (and the small change below).